### PR TITLE
copy from

### LINKED ISSUE #2946: Develop "import from" on entity creation / edit
Instead of the "duplicate entity" procedure requested on #2793 , we are moving on with "import from".

This will allow to, while creating or editing an entity, import matching template properties from a different entity.

### PR DESCRIPTION
**Pull request by konzz**
fixes #2946 

PR checklist:
- [ ] Update READ.me ?
- [ ] Update API documentation ?

QA checklist:
- [ ] Smoke test the functionality described in the issue
- [ ] Test for side effects
- [ ] UI responsiveness
- [ ] Cross browser testing
- [ ] Code review

### CODE REVIEW

File: app/react/Metadata/components/specs/CopyFromEntity.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  describe('render', () => {
+    it('should render a search entities component with onSelect callback', async () => {
+      render();
+      const searcher = component.find(SearchEntities);
+      expect(searcher.props().onSelect).toBe(component.instance().onSelect);
+    });
+  });
+
+  describe('when an entity is selected', () => {
+    it('should render the entity an set the comon props in the satet', () => {
```
Comment (fnocetti) [inline app/react/Metadata/components/specs/CopyFromEntity.spec.tsx:59]: There's two typos here: comon should be common and satet should be state :)

File: app/react/Metadata/components/specs/SearchEntities.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+  const render = () => {
+    component = shallow(<SearchEntities {...props} />);
+  };
+
+  describe('search', () => {
+    it('should request for the entities matching by title', async () => {
+      render();
+      component.find(SearchInput).simulate('change', { target: { value: 'test' } });
+      await sleep(4001);
```
Comment (fnocetti) [inline app/react/Metadata/components/specs/SearchEntities.spec.tsx:39]: You know this better than me, but is this number arbitrary?

File: app/react/App/scss/layout/_sidepanel.scss [hunk 3]
```diff
@@ -20,13 +20,38 @@ $c-sidebar: $c-white;
...
   }
 
+  &.two-columns #metadataForm {
+    display: none;
+  }
+
+  @media (min-width: 700px) {
+    &.two-columns {
+      width: 700px;
+      -webkit-box-shadow: -4px 0px 10px 0px rgba(0, 0, 0, 0.5);
```
[the remark this answers is not in the document]
Comment (RafaPolit) [reply app/react/App/scss/layout/_sidepanel.scss:37]: I believe @fnocetti  is right.  `box-shadow` will get correctly replicated with the different specific values.

**Comment by fnocetti** [comment]
@konzz I'm not being able to copy metadata. Here are the repo steps:

- Create template A
- Create entity tA of type A
- Start creation of entity cA of type A
  - Select Copy From
  - Search for entity tA
  - Click Copy Highlighted

Result: It goes back to the entity creation sidebar, but no content is copied into the fields. What's more, the form it's been reseted (for example, the chosen template is changed back to the default).

May I be doing something wrong? Can you please check?

File: app/react/Metadata/components/specs/SearchEntities.spec.tsx (hunk 2 above)
Comment (konzz) [reply app/react/Metadata/components/specs/SearchEntities.spec.tsx:39]: I've changed it to 401, but it is related to the debounce time that is in the search function that has 400ms

Comment (konzz) [reply app/react/Metadata/components/specs/SearchEntities.spec.tsx:39]: I've changed it to 401ms, but is related to the debounce time that the search function has.

**Comment by konzz** [comment]
Yep there was a bug when doing copy from on new entities.

**Comment by fnocetti** [comment]
LGTM!

**Pull request merged**